aten/src/ATen/native/cuda/TensorShape.cu

```diff
@@ -713,7 +744,13 @@ Tensor _chunk_cat_cuda(TensorList tensors, int64_t dim, int64_t num_chunks) {
          leading_dim,
          num_blocks_per_chunk,
          slice_size,
-         device_ptrs] =
+         srcs,
+         block_idx_to_tensor_idx,
```

Comment:
It seems that you are only using `chunk_size` and `leading_dim` here, but you introduced 9 extra unused names by calling `detail::get_chunk_cat_metadata`. Let's just refactor `chunk_size` and `leading_dim` calculation into separate functions.